### PR TITLE
Support Influx V2 on Docker

### LINKED ISSUE #67: docker support for influx v2
Currently the docker deployment uses influxdb v1, which should be replaced with influx v2 and necessary config should be updated.

### PR DESCRIPTION
**Pull request by AarC10**
# Description
Adds support for InfluxV2 on Docker

# How Has This Been Tested?
Ran the docker compose and ran some queries on Influx through docker

# Checklist:
- [x] New functionality is documented in the necessary spots (i.e new functions documented in the header)
- [x] Unit tests cover any new functionality or edge cases that the PR was meant to resolve (if applicable) 
- [x] The CI checks are passing
- [x] I reviewed my own code in the GitHub diff and am sure that each change is intentional

### CODE REVIEW

File: data/influxdb/init-v1-compat.sh
Comment (AarC10) [inline data/influxdb/init-v1-compat.sh:1]: Note this file should be blasted once we blast influx v1 from existence. This is just a bandaid to maintain compatibility with the current Grafana dashboards (technically can blast when that happens actually)

File: compose-services.yaml [hunk 1]
```diff
@@ -6,26 +6,52 @@ services:
...
+      - GSW_GRAFANA_INFLUXDB_DB_NAME=${GSW_INFLUXDB_BUCKET:-gsw}
+      - GSW_GRAFANA_INFLUXDB_TOKEN=${GSW_INFLUXDB_ADMIN_TOKEN:-gsw-local-dev-token}
       - GF_AUTH_ANONYMOUS_ENABLED=true
       - GF_AUTH_ANONYMOUS_ORG_ROLE=Viewer
       - GF_AUTH_BASIC_ENABLED=false
     volumes:
       - grafana-data:/var/lib/grafana
   influxdb:
-    image: influxdb:1.11-alpine
+    image: influxdb:2.7-alpine
```
Comment (K3das) [inline compose-services.yaml:21]: `2.8-alpine` is latest, could probably be bumped unless there are breaking changes in 2.8.

File: README.md [hunk 2]
```diff
@@ -57,12 +57,17 @@ If you need to change dashboards or datasources, set `GF_AUTH_ANONYMOUS_ORG_ROLE
...
-    -format csv \
-    -execute "SELECT * FROM receiver"
+docker compose exec influxdb influx bucket list
+```
+
+To query the `gsw` bucket directly, run:
+```shell
+docker compose exec influxdb influx query \
+    --org gsw \
+    --token gsw-local-dev-token \
```
[the remark this answers is not in the document]
Comment (K3das) [reply README.md:69]: hardcoded token is probably fine in the docs

File: README.md [hunk 3]
```diff
@@ -57,12 +57,17 @@ If you need to change dashboards or datasources, set `GF_AUTH_ANONYMOUS_ORG_ROLE
...
+    'from(bucket: "gsw") |> range(start: -15m)'
```
Comment (K3das) [inline README.md:70]: this could potentially be changed to `start: 0` to maintain compatibility with previous docs but it's like no big deal

File: data/influxdb/init-v1-compat.sh
Comment (K3das) [reply data/influxdb/init-v1-compat.sh:1]: honestly wouldn't hurt to have this in general

Comment (K3das) [inline data/influxdb/init-v1-compat.sh:1]: Could this be moved to an init script in the influx container? Would make it a lot simpler on the docker-compose and config side of things.

https://docs.influxdata.com/influxdb/v2/install/use-docker-compose/#:~:text=Runs%20any%20custom%20initialization%20scripts%20mounted%20inside%20the%20container%E2%80%99s%20/docker%2Dentrypoint%2Dinitdb.d/%20path.

File: cmd/Containerfile [hunk 4]
```diff
@@ -43,8 +43,10 @@ USER appuser
 
 ENV GSW_LOGGER_OUTPUT_PATHS=stdout
 
-ENV GSW_DATABASE_HOST_NAME=influxdb
-ENV GSW_DATABASE_PORT_NUMBER=8089
+ENV GSW_DATABASE_V2_URL=http://influxdb:8086
+ENV GSW_DATABASE_V2_TOKEN=gsw-local-dev-token
+ENV GSW_DATABASE_V2_ORG=gsw
+ENV GSW_DATABASE_V2_BUCKET=gsw
```
Comment (K3das) [inline cmd/Containerfile:49]: I believe I had the database environment variables in the Dockerfile so that the GSW container could start without a database config, but that was patched previously so now I think all of the database config could be left out of the Dockerfile.

**Review by K3das — CHANGES_REQUESTED** [review]
lgtm- just some slightly subjective changes.

**Review by K3das — APPROVED** [review]
